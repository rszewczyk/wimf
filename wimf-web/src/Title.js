// @flow
import React from "react";
import { css } from "glamor";

type Size = 1 | 2 | 3 | 4 | 5 | 6;

type TitleProps = {
  title: string,
  description?: string,
  size: Size
};

const titleStyle = css({
  marginBottom: "1.0rem",
  display: "flex",
  flexDirection: "column",
  " h1, h2, h3, h4, h5, h6": {
    display: "inline-block"
  }
});

export default function(props: TitleProps) {
  const { title, description, size } = props;

  return (
    <div {...titleStyle}>
      {React.createElement(`h${size || 3}`, null, title)}
      {description && <div>{description}</div>}
    </div>
  );
}
