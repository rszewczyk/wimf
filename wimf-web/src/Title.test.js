import React from "react";
import renderer from "react-test-renderer";
import Title from "./Title";

it("renders correctly", () => {
  const withoutDescription = renderer
    .create(<Title title="some title" />)
    .toJSON();
  expect(withoutDescription).toMatchSnapshot();

  const withDescription = renderer
    .create(<Title title="some title" description="some description" />)
    .toJSON();
  expect(withDescription).toMatchSnapshot();
});
