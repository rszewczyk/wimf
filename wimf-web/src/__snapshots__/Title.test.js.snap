// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  data-css-7q9pzb=""
>
  <h3>
    some title
  </h3>
</div>
`;

exports[`renders correctly 2`] = `
<div
  data-css-7q9pzb=""
>
  <h3>
    some title
  </h3>
  <div>
    some description
  </div>
</div>
`;
